crates/factor-outbound-pg/src/client.rs

```diff
@@ -0,0 +1,75 @@
+use native_tls::TlsConnector;
+use postgres_native_tls::MakeTlsConnector;
+use spin_world::async_trait;
+use tokio_postgres::{config::SslMode, types::ToSql, Error, Row, ToStatement};
+use tokio_postgres::{Client as PgClient, NoTls, Socket};
+
+#[async_trait]
+pub trait Client {
+    async fn build_client(address: &str) -> anyhow::Result<Self>
+    where
+        Self: Sized;
+
+    async fn execute<T>(&self, query: &T, params: &[&(dyn ToSql + Sync)]) -> Result<u64, Error>
+    where
+        T: ?Sized + ToStatement + Sync + Send;
+
+    async fn query<T>(&self, query: &T, params: &[&(dyn ToSql + Sync)]) -> Result<Vec<Row>, Error>
```

Comment:
Seems strange that we're forcing all implementors to return a `Row` type that is defined in `tokio_postgres` (along with other uses of `tokio_postgres` types). I would have assumed we would define this crate in terms of the wit types. This would mean conversion from implementation specific types to wit types would be contained inside the `Client` impl which makes sense to me.

Reply:
I think you're probably right. I spent some time looking at how I would do this and I was stumped by one thing though. The `v1` and `v2` interfaces have a slightly different shape (return different errors and take different params). How should I reconcile that in a single client interface?

Reply:
Simply by mapping the params and return values. We do this in SQLite: https://github.com/fermyon/spin/blob/14f8a5897f17c7cfc053d2b9f59e1843b6c9a72e/crates/factor-sqlite/src/host.rs#L108-L116